modules/enterprise/server/server-metrics/src/main/java/org/rhq/server/metrics/aggregation/DataAggregator.java

```diff
@@ -351,8 +351,10 @@ public AggregationTask(Batch batch) {
         @Override
         public void run() {
             try {
+                permits.acquire();
                 run(batch);
             } catch (Exception e) {
+                permits.release();
```

Comment:
This can't be right? Only in the case of Exception would you release something here (and then in one FutureCallback finally blocks one more semaphore release). Thus a single acquire, two releases..